dbms/src/Flash/Statistics/ExecutorStatisticsCollector.cpp

```diff
@@ -65,22 +80,121 @@ void ExecutorStatisticsCollector::initialize(DAGContext * dag_context_)
                 TableScanStatistics,
                 TopNStatistics,
                 WindowStatistics,
-                ExpandStatistics>(executor_id, &executor))
+                ExpandStatistics>(&executor))
         {
             throw TiFlashException(
-                fmt::format("Unknown executor type, executor_id: {}", executor_id),
+                fmt::format("Unknown executor type, executor_id: {}", executor.executor_id()),
                 Errors::Coprocessor::Internal);
         }
         return true;
     });
+
+    fillListBasedExecutorsChild();
+}
+
+void ExecutorStatisticsCollector::fillListBasedExecutorsChild()
+{
+    if (!dag_context->return_executor_id)
```

Comment:
Maybe we can do this judge in a new function and the reader could get the purpose of this judgement by function name.